c_src/erl_optics.c

```diff
@@ -18,6 +18,8 @@ static ERL_NIF_TERM atom_dist;
 static ERL_NIF_TERM atom_quantile;
 static ERL_NIF_TERM atom_histo;
 static struct optics_poller *poller;
+static bool carbon_poller;
+static bool erlang_poller;
```

Comment:
Because I'm paranoid, I always initialize my static variables and I would recommend you do the same. Note that accessing unitialized variable is undefined behavior which is problematic for global variables which can be accessed from anywhere at any time.

Reply:
Done.